### DISCUSSION TITLE
How to Fix  Capturing groups are not allowed  in rewirtes function in next.config.js

### DISCUSSION THREAD

**Discussion by wildfrontend** [vercel/next.js#19311] [Help] [answered]
```
module.exports = {
  rewrites: async () => [
    {
      source: '/match/:gameType(^(?!(baseball|basketball|soccer)$))',
      destination: '/**',
    },
  ]
}
```
error message
![截圖 2020-11-19 下午7 40 05](https://user-images.githubusercontent.com/20965813/99661589-06f7c580-2a9f-11eb-8b49-f290f6cde295.png)

**Comment by ijjk** [comment, marked as answer]
Hi, it looks like you need to update your regex a bit for `path-to-regexp`, we have [this error](https://github.com/vercel/next.js/blob/13f8732f86ad449366f29a843af32379be72cacc/errors/invalid-route-source.md) doc that can sometimes show if we notice the issue you're running into although it doesn't seem to be triggered for this case. 

Updating the above source to: `/match/:gameType((?!baseball$|basketball$|soccer$).*)` should work correctly

**Comment by Officer-R** [comment]
@wildfrontend I am facing a similar issue while on a gatsby project.

gatsby-plugin-react-i18next plugin is using `path-to-regexp` for matching the path => https://www.gatsbyjs.com/plugins/gatsby-plugin-react-i18next/#page-options

<img width="1439" alt="image" src="https://user-images.githubusercontent.com/37110560/174302777-68ef2f79-e9d8-468a-8df7-bb64f1b9bb4d.png">


The pattern which i am trying to match => `"/:lang?/bali/:rest(about|locate-us|offers|wellness)((\/))?(([?][a-z_=A-Z]+))?(([a-z-])+\/)?"`
